### PR TITLE
Improve performance of standalone commands by starting the event monitor on-demand

### PR DESCRIPTION
**Pull request by romac**
Closes: #1063

## Description

Lazily spawn the event monitor and subscribe to events over WebSocket only when `ChainHandle::subscribe()` is called, which improves the startup time of commands which do not need to subscribe to WS events.

______

For contributor use:

- [x] Added a changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer/src/chain/runtime.rs [hunk 1]
```diff
@@ -1,5 +1,5 @@
 use alloc::sync::Arc;
-use std::thread;
+use std::{option::Option, thread};
```
Comment (romac) [inline relayer/src/chain/runtime.rs:2]:
```suggestion
use std::thread;
```

**Comment by mzabaluev** [comment]
Needs rebasing.

File: relayer/src/event/monitor/error.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            |_| { "RPC error" },
+    }
+}
+
+impl Error {
+    pub fn canceled_or_generic(e: RpcError) -> Self {
+        use tendermint_rpc::error::ErrorDetail;
+
+        match e.detail() {
+            ErrorDetail::Server(detail) if detail.reason.contains("subscription was cancelled") => {
```
Comment (mzabaluev) [inline relayer/src/event/monitor/error.rs:56]: Can we demand a more type-checked way to identify this reason from `tendermint-rpc`?
Matching by string is liable to silently stopping to work, unless the tendermint-rpc developers commit to stable reason strings as part of their API.

Comment (romac) [reply relayer/src/event/monitor/error.rs:56]: This error is actually emitted as a string by the Cosmos SDK, so there's not much we can do here aside from moving that very same logic into `tendermint-rpc`.

**Review by mzabaluev — APPROVED** [review]
Looks good, other than issues outside of this project's control (I've got to raise an issue against cosmos-sdk).

**Pull request merged**